unit/util/piped_process.cpp

```diff
@@ -0,0 +1,272 @@
+/// \file
+/// \author Diffblue Ltd.
+/// Unit tests for checking the piped process communication mechanism.
+
+#ifdef _WIN32
+// No unit tests yet!
+#else
+
+#  include <testing-utils/use_catch.h>
+#  include <util/optional.h>
+#  include <util/piped_process.h>
+#  include <util/string_utils.h>
+
+TEST_CASE(
+  "Creating a sub process and reading its output.",
+  "[core][util][piped_process]")
+{
+  const std::string to_be_echoed = "The Jabberwocky";
+  // Need to give path to avoid shell built-in invocation
+  std::vector<std::string> commands;
+  commands.push_back("/bin/echo");
+  commands.push_back(to_be_echoed);
+  piped_processt process = piped_processt(commands);
```

Comment:
:bulb: The command vector could be initialised inline in order to avoid mutating a vector with `push_back`. For example - `piped_processt process{{"/bin/echo", to_be_echoed}};`